From 595b84d71c477ae5b6a55e573b030911e5410f24 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:03:58 -0400
Subject: [net] usb/cdc_ncm: remove tx_speed and rx_speed fields

Message-id: <1410887463-20674-167-git-send-email-dzickus@redhat.com>
Patchwork-id: 95134
O-Subject: [RHEL7 PATCH 166/591] net: cdc_ncm: remove tx_speed and rx_speed fields
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit f3028c524a7cd4d97b034fc1f35dcaecb5d6f9d6
Author: Bjorn Mork <bjorn@mork.no>
Date:   Fri Nov 1 11:16:44 2013 +0100

    net: cdc_ncm: remove tx_speed and rx_speed fields

    These fields are only used to prevent printing the same speeds
    multiple times if we receive multiple identical speed notifications.

    The value of these printk's is questionable, and even more so when
    we filter out some of the notifications sent us by the firmware. If
    we are going to print any of these, then we should print them all.

    Removing little used fields is a bonus.

    Cc: Alexey Orishko <alexey.orishko@gmail.com>
    Signed-off-by: Bjorn Mork <bjorn@mork.no>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/usb/cdc_ncm.c b/drivers/net/usb/cdc_ncm.c
index 9cdd762..fc36a99 100644
--- a/drivers/net/usb/cdc_ncm.c
+++ b/drivers/net/usb/cdc_ncm.c
@@ -510,7 +510,6 @@ advance:
      ctx->tx_max % usb_maxpacket(dev->udev, dev->out, 1) == 0)
   ctx->tx_max++;
 
- ctx->tx_speed = ctx->rx_speed = 0;
  return 0;
 
 error2:
@@ -1048,7 +1047,6 @@ static void
 cdc_ncm_speed_change(struct usbnet *dev,
        struct usb_cdc_speed_change *data)
 {
- struct cdc_ncm_ctx *ctx = (struct cdc_ncm_ctx *)dev->data[0];
  uint32_t rx_speed = le32_to_cpu(data->DLBitRRate);
  uint32_t tx_speed = le32_to_cpu(data->ULBitRate);
 
@@ -1056,25 +1054,20 @@ cdc_ncm_speed_change(struct usbnet *dev,
   * Currently the USB-NET API does not support reporting the actual
   * device speed. Do print it instead.
   */
- if ((tx_speed != ctx->tx_speed) || (rx_speed != ctx->rx_speed)) {
-  ctx->tx_speed = tx_speed;
-  ctx->rx_speed = rx_speed;
-
-  if ((tx_speed > 1000000) && (rx_speed > 1000000)) {
-   printk(KERN_INFO KBUILD_MODNAME
-          ": %s: %u mbit/s downlink "
-          "%u mbit/s uplink\n",
-          dev->net->name,
-          (unsigned int)(rx_speed / 1000000U),
-          (unsigned int)(tx_speed / 1000000U));
-  } else {
-   printk(KERN_INFO KBUILD_MODNAME
-          ": %s: %u kbit/s downlink "
-          "%u kbit/s uplink\n",
-          dev->net->name,
-          (unsigned int)(rx_speed / 1000U),
-          (unsigned int)(tx_speed / 1000U));
-  }
+ if ((tx_speed > 1000000) && (rx_speed > 1000000)) {
+  printk(KERN_INFO KBUILD_MODNAME
+         ": %s: %u mbit/s downlink "
+         "%u mbit/s uplink\n",
+         dev->net->name,
+         (unsigned int)(rx_speed / 1000000U),
+         (unsigned int)(tx_speed / 1000000U));
+ } else {
+  printk(KERN_INFO KBUILD_MODNAME
+         ": %s: %u kbit/s downlink "
+         "%u kbit/s uplink\n",
+         dev->net->name,
+         (unsigned int)(rx_speed / 1000U),
+         (unsigned int)(tx_speed / 1000U));
  }
 }
 
@@ -1111,8 +1104,6 @@ static void cdc_ncm_status(struct usbnet *dev, struct urb *urb)
          dev->net->name, ctx->connected ? "" : "dis");
 
   usbnet_link_change(dev, ctx->connected, 0);
-  if (!ctx->connected)
-   ctx->tx_speed = ctx->rx_speed = 0;
   break;
 
  case USB_CDC_NOTIFY_SPEED_CHANGE:
diff --git a/include/linux/usb/cdc_ncm.h b/include/linux/usb/cdc_ncm.h
index 059dcc9..f14af3d 100644
--- a/include/linux/usb/cdc_ncm.h
+++ b/include/linux/usb/cdc_ncm.h
@@ -110,8 +110,6 @@ struct cdc_ncm_ctx {
 
  u32 tx_timer_pending;
  u32 tx_curr_frame_num;
- u32 rx_speed;
- u32 tx_speed;
  u32 rx_max;
  u32 tx_max;
  u32 max_datagram_size;
-- 
1.7.1